src/test/java/net/minecraftforge/test/GuiTabsTest.java

```diff
@@ -0,0 +1,221 @@
+                return new ItemStack(Item.getItemFromBlock(Blocks.ICE));
+            }
+        };
+    }
+
+    public static class CommonProxy
+    {
+        public void init()
+        {
+            NetworkRegistry.INSTANCE.registerGuiHandler(GuiTabsTest.instance, new GuiHandler());
+            GuiTabsTest.INSTANCE.registerMessage(TabMessageHandler.class, TabMessageHandler.TabMessage.class, 0, Side.SERVER);
+        }
+    }
+
+    public static class ServerProxy extends CommonProxy
+    {
+        public void init()
+        {
+            super.init();
+        }
+    }
+
+    public static class ClientProxy extends CommonProxy
+    {
+        public void init()
+        {
+            super.init();
+            new GuiTab("Tab1", new ItemStack(Blocks.BEDROCK), TestGui.class)
+            {
+                @Override
+                public void onTabClicked(GuiContainer guiContainer)
+                {
+                    super.onTabClicked(guiContainer);
+                    GuiTabsTest.INSTANCE.sendToServer(new TabMessageHandler.TabMessage(0));
+                }
+            }.addTo(GuiInventory.class);
+            new GuiTab("Tab2", new ResourceLocation("textures/blocks/tnt_side.png"), GuiChest.class)
+            {
+                @Override
+                public void onTabClicked(GuiContainer guiContainer)
+                {
+                    super.onTabClicked(guiContainer);
+                    GuiTabsTest.INSTANCE.sendToServer(new TabMessageHandler.TabMessage(1));
+                }
+            }.addTo(GuiFurnace.class);
+            ItemStack is = new ItemStack(Items.APPLE);
+            is.addEnchantment(new net.minecraft.enchantment.EnchantmentProtection(Enchantment.Rarity.COMMON, EnchantmentProtection.Type.FALL, null), 1);
+            new GuiTab("Tab3", is, TestGui2.class)
+            {
+                @Override
+                public void onTabClicked(GuiContainer guiContainer)
+                {
+                    super.onTabClicked(guiContainer);
+                    GuiTabsTest.INSTANCE.sendToServer(new TabMessageHandler.TabMessage(2));
+                }
+            }.addTo(GuiInventory.class);
+        }
+
+    }
+
+    public static class TestGui extends GuiChest
+    {
+        public TestGui(IInventory upperInv, IInventory lowerInv)
+        {
+            super(upperInv, lowerInv);
+        }
+    }
+
+    public static class TestGui2 extends TestGui
+    {
+        public TestGui2(IInventory upperInv, IInventory lowerInv)
+        {
+            super(upperInv, lowerInv);
+        }
+    }
+
+    public static class GuiHandler implements IGuiHandler
+    {
+
+        IInventory invC;
+        IInventory invS;
+
+        @Override
+        public Object getServerGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z)
+        {
+            if (invS == null)
+                invS = new InventoryBasic("Test", true, 27);
+            switch (ID)
+            {
+            default:
```

Comment:
This indentation is weird...

Reply:
*shrug* used: https://github.com/ForgeEssentials/ForgeEssentials/blob/develop/misc/intellijformat.xml
but I guess that switch is redundant

Reply:
It is fine. It is a test mod, which no one really cares.